### PR TITLE
feat(openai): Add support for custom base_url and model

### LINKED ISSUE #21: Add support for custom base_url and model for OpenAI embedding
Edgen added support for open ai compatible apis for embedding https://github.com/edgenai/edgen/pull/105

Ollama has a PR for it https://github.com/ollama/ollama/pull/2925

This will make it easy to get started with self hosted RAGs.

### PR DESCRIPTION
**Pull request by Abraxas-365**
Add support for custom base_url and model for OpenAI embedding. This change allows users to specify their own base URL and model when using the OpenAiEmbedder struct. This will make it easy to get started with self-hosted RAGs.

Closes #21

### CODE REVIEW

**Comment by Abraxas-365** [comment]
Sorry , i close it by accident

**Comment by prabirshrestha** [comment]
LGTM. You can also add support for `OPENAI_API_BASE` env variable.

Probably worth investing in builder apis.

File: src/embedding/openai/openai_embedder.rs [hunk 1]
```diff
@@ -37,24 +37,41 @@ struct UsageData {
...
 
 impl Default for OpenAiEmbedder {
     fn default() -> Self {
-        OpenAiEmbedder {
-            model: String::from("text-embedding-ada-002"),
-            openai_key: env::var("OPENAI_API_KEY").unwrap_or(String::new()),
-        }
+        let model = String::from("text-embedding-ada-002");
+        let openai_key = env::var("OPENAI_API_KEY").unwrap_or(String::new());
+        let base_url = String::from("https://api.openai.com/v1/embeddings");
```
Comment (prabirshrestha) [inline src/embedding/openai/openai_embedder.rs:73]: add support for OPENAI_API_BASE?

**Review by prabirshrestha — APPROVED** [review]

**Comment by Abraxas-365** [comment]
> LGTM. You can also add support for `OPENAI_API_BASE` env variable.
> 
> Probably worth investing in builder apis.

Can you crate issue for the OPENAI_API_BASE pls. should it be just the bese https://api.openai.com ? becuase the v1/embedding also could change too. are Edger and  openai using the same v1/embedding?

**Pull request merged**